diff --git a/rc/filetype/janet.kak b/rc/filetype/janet.kak
index 00f39c2..9d30b38 100644
--- a/rc/filetype/janet.kak
+++ b/rc/filetype/janet.kak
@@ -72,7 +72,7 @@ define-command -hidden janet-indent-on-new-line %{
         }
         try %{ execute-keys -draft '[rl"i<a-Z><gt>' }
         try %{ execute-keys -draft '[Bl"i<a-Z><gt>' }
-        execute-keys -draft '"i<a-z>a&,'
+        execute-keys -draft ';"i<a-z>a&,'
         # trim trailing whitespace on the previous line
         try %{ execute-keys -draft k : janet-trim-indent <ret> }
     }
